plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/statistics/CachingExtendedStatisticsAccess.java

```diff
@@ -68,32 +70,47 @@ public Optional<ExtendedStatistics> readExtendedStatistics(ConnectorSession sess
     }
 
     @Override
-    public void updateExtendedStatistics(ConnectorSession session, String tableLocation, ExtendedStatistics statistics)
+    public void updateExtendedStatistics(ConnectorSession session, SchemaTableName schemaTableName, String tableLocation, ExtendedStatistics statistics)
     {
-        delegate.updateExtendedStatistics(session, tableLocation, statistics);
-        cache.invalidate(tableLocation);
+        delegate.updateExtendedStatistics(session, schemaTableName, tableLocation, statistics);
+        cache.invalidate(new CacheKey(schemaTableName, tableLocation));
     }
 
     @Override
-    public void deleteExtendedStatistics(ConnectorSession session, String tableLocation)
+    public void deleteExtendedStatistics(ConnectorSession session, SchemaTableName schemaTableName, String tableLocation)
     {
-        delegate.deleteExtendedStatistics(session, tableLocation);
-        cache.invalidate(tableLocation);
+        delegate.deleteExtendedStatistics(session, schemaTableName, tableLocation);
+        cache.invalidate(new CacheKey(schemaTableName, tableLocation));
     }
 
     public void invalidateCache()
     {
         cache.invalidateAll();
     }
 
-    public void invalidateCache(String tableLocation)
+    // for explicit cache invalidation
+    public void invalidateCache(SchemaTableName schemaTableName, Optional<String> tableLocation)
     {
-        // for explicit cache invalidation
-        cache.invalidate(tableLocation);
+        requireNonNull(schemaTableName, "schemaTableName is null");
+        if (tableLocation.isPresent()) {
+            cache.invalidate(new CacheKey(schemaTableName, tableLocation.get()));
```

Comment:
I think you want to invalidate based on location and name each separately here, even if the name is present. It is probably rare but an external system could drop and recreate a table using the same location but a different name, or same name in a different location.

Reply:
for this to matter, one location would need to be used interchangibly with different table names, right?

Reply:
applied. note that this way of invalidation will be slower (linear search)